src/main/java/com/hartwig/platinum/config/version/PipelineVersion.java

```diff
@@ -0,0 +1,28 @@
+package com.hartwig.platinum.config.version;
+
+import static java.lang.String.format;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PipelineVersion {
+    String pad(final String candidate) {
+        ArrayList<String> versionsGiven = new ArrayList<>(List.of(candidate.split("\\.")));
+        int i;
+        int numMissing = 3 - versionsGiven.size();
+        for (i = 0; i < numMissing; i++) {
+            versionsGiven.add("0");
+        }
+        return String.join(".", versionsGiven);
+    }
+
+    String extract(final String fullPipelineVersion) {
```

Comment:
Maybe rename method to something slightly specific IE `extractVersionFromDockerImageName`

Reply:
Nice suggestion. Corrected.